lib/auth/github.go

```diff
@@ -962,25 +985,39 @@ func ValidateClientRedirect(clientRedirect string, ssoTestFlow bool, settings *t
 			return trace.AccessDenied("custom client redirect URLs are not allowed in SSO test")
 		}
 
-		switch u.Port() {
-		default:
-			return trace.BadParameter("invalid port in client redirect URL")
-		case "", "443":
+		if settings == nil || (settings.AllowedHttpsHostnames == nil && settings.InsecureAllowedCidrRanges == nil) {
+			return trace.BadParameter("invalid hostname in client redirect URL")
 		}
 
-		var allowedHostnames []string
-		if settings != nil {
-			allowedHostnames = settings.AllowedHttpsHostnames
+		hostIP, err := netip.ParseAddr(u.Hostname())
+		if err == nil {
+			for _, cidrStr := range settings.InsecureAllowedCidrRanges {
```

Comment:
I don't like how we have to duplicate this check for http and https.

What if we did it like
```
if scheme isn't http or https fail
if http and localhost, 127.0.0.1 or ::1 ok
if hostname is an address and is contained in InsecureAllowedCidrRanges ok
if not https fail
if port is not default or 443 fail
if hostname matches AllowedHttpsHostnames ok
fail
```